src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -583,7 +650,19 @@ static BoundStatement generateIndexer(SyntheticBoundNodeFactory f, MethodSymbol
                 var field = containingType._field;
                 var fieldReference = f.Field(f.This(), field);
                 var parameterReference = f.Parameter(method.Parameters[0]);
-                if (field.Type.IsArray())
+                if (containingType.IsSingleton)
+                {
+                    // if (param0 != 0)
+                    //      throw new IndexOutOfRangeException();
+                    // return _item;
+                    var constructor = (MethodSymbol)method.DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_IndexOutOfRangeException__ctor)!;
```

Comment:
>WellKnownMember.System_IndexOutOfRangeException__ctor [](http://example.com/codeflow?start=103&length=53)

Let's include `WellKnownMember.System_IndexOutOfRangeException__ctor` in the set of members to check for in `Create` for `SynthesizedReadOnlyListKind.Singleton`, and include in the corresponding test for missing members.